### PR TITLE
Changed format of employee contact information so the address conatai…

### PR DESCRIPTION
**Pull request by Felixim0**
…ns all associated address information such as postcode.

### CODE REVIEW

File: app/fieldmapping.py [hunk 1]
```diff
@@ -88,6 +88,18 @@ def map_employee_history_job_role_table_headers(
...
+  complete_address = ''
+  address_parts  = [
+      'address1',
+      'address2',
+      'town',
+      'country',
+      'postcode',
+      ]
+  for address_part in address_parts:
+    complete_address = complete_address + str( employee_info.get(address_part,'') ) + ",  "
```
Comment (SammyPoot) [inline app/fieldmapping.py]: The end of this line looks like it could use reformatting.

Comment (SammyPoot) [inline app/fieldmapping.py]: The logic here seems odd, and will leave a trailing comma+space at the end of the joining.
Python provides a method for this kind of situation: `string.join(iterable)`
In this case, `complete_address = ", ".join(part + str(employee_info.get(part, '') for part in address_parts)` might give you the full string in one line.

File: app/views/fsss_view.py [hunk 2]
```diff
@@ -54,6 +54,8 @@ def get_emp_info(name, on_false={}, on_missing='Unspecified'):
 
   preferred_name = get_emp_info('preferredName', on_false='None')
 
+  employee_info['address'] = map_full_address_details(employee_info)
```
Comment (SammyPoot) [inline app/views/fsss_view.py]: Mutating one of the inputs in this way is a solution that's prone to error. Unless you check the entire function, you cannot know that `employee_info` was tampered with in this way.

A more reliable system would be to take `'address'` out of the `EMP_CONTACT_FIELDS` dictionary, and manually add the address to `data_contact = {}`, by changing it to `data_contact = {'Address': map_full_address_details(employee_info)}`

File: app/views/logistics_view.py [hunk 3]
```diff
@@ -22,6 +22,8 @@ def get_emp_info(name, on_false={}, on_missing='Unspecified'):
 
   mob_number = device_numbers[0] if len(device_numbers) > 0 else ''
 
+  employee_info['address'] = map_full_address_details(employee_info)
```
Comment (SammyPoot) [inline app/views/logistics_view.py]: Ditto, as with the above comment on input mutation.

File: app/views/hr_view.py [hunk 4]
```diff
@@ -56,6 +56,8 @@ def get_emp_info(name, on_false={}, on_missing='Unspecified'):
 
   line_manager = format_line_manager(current_job_role)
 
+  employee_info['address'] = map_full_address_details(employee_info)
```
Comment (SammyPoot) [inline app/views/hr_view.py]: Ditto, as with the above comment on input mutation.

File: app/views/rmt_view.py [hunk 5]
```diff
@@ -21,6 +21,8 @@ def get_emp_info(name, on_false={}, on_missing='Unspecified'):
 
   mob_number = device_numbers[0] if len(device_numbers) > 0 else ''
 
+  employee_info['address'] = map_full_address_details(employee_info)
```
Comment (SammyPoot) [inline app/views/rmt_view.py]: Ditto, as with the above comment on input mutation.

**Review by SammyPoot — CHANGES_REQUESTED** [review]

File: app/fieldmapping.py (hunk 1 above)
Comment (Felixim0) [reply app/fieldmapping.py]: Changed

File: app/views/fsss_view.py (hunk 2 above)
Comment (Felixim0) [reply app/views/fsss_view.py]: Have changed this in all the views

File: app/views/rmt_view.py (hunk 5 above)
Comment (Felixim0) [reply app/views/rmt_view.py]: Changed too

File: app/views/fsss_view.py [hunk 6]
```diff
@@ -92,7 +89,10 @@ def get_emp_info(name, on_false={}, on_missing='Unspecified'):
     data_job_role[mapField] = get_emp_info(empField)
   tab_job_role = tab_generation('Job Role for Field Worker', data_job_role)
 
-  data_contact = {}
+  data_contact = {
+      'Address': map_full_address_details(employee_info)
+      }
```
Comment (SammyPoot) [inline app/views/fsss_view.py]: The formatting on this is off- please re-run the formatter before each commit.

File: app/fieldmapping.py [hunk 7]
```diff
@@ -88,6 +88,19 @@ def map_employee_history_job_role_table_headers(
...
 
+def map_full_address_details(employee_info):
+  complete_address = ''
+  address_parts  = [
+      'address1',
+      'address2',
+      'town',
+      'country',
+      'postcode',
+      ]
```
Comment (SammyPoot) [inline app/fieldmapping.py]: Please run the reformatter before each commit.

**Review by SammyPoot — CHANGES_REQUESTED** [review]

File: app/fieldmapping.py [hunk 8]
```diff
@@ -88,6 +88,20 @@ def map_employee_history_job_role_table_headers(
 
   return table_generation(mapping_entries)
 
+def map_full_address_details(employee_info):
+  complete_address = ''
```
Comment (SammyPoot) [inline app/fieldmapping.py]: This line isn't needed at all. Don't initialize any variables that you're going to fully replace later.

File: app/views/hr_view.py (hunk 4 above)
Comment (SammyPoot) [reply app/views/hr_view.py]: This hasn't been resolved in the past commits.

File: app/views/rmt_view.py [hunk 9]
```diff
@@ -1,6 +1,6 @@
 from app.employee_view_functions import process_device_details, format_line_manager
 from app.tabutils import tab_generation, format_to_uk_dates
-from app.fieldmapping import map_employee_name
+from app.fieldmapping import map_employee_name, map_full_address_details
```
Comment (SammyPoot) [inline app/views/rmt_view.py]: The function isn't used at all in this file.

**Review by SammyPoot — CHANGES_REQUESTED** [review]

File: app/views/hr_view.py [hunk 10]
```diff
@@ -1,6 +1,6 @@
 from app.employee_view_functions import process_device_details, format_line_manager, extract_device_phone, extract_device_chromebook
 from app.tabutils import tab_generation, format_to_uk_dates
-from app.fieldmapping import map_employee_name
+from app.fieldmapping import map_employee_name, map_full_address_details
```
Comment (SammyPoot) [inline app/views/hr_view.py]: The function isn't used in this file.

File: app/views/fsss_view.py (hunk 6 above)
Comment (SammyPoot) [reply app/views/fsss_view.py]: Do you need to add a trailing comma to the end of the Address line? It's still indenting the ending bracket.

**Review by SammyPoot — CHANGES_REQUESTED** [review]

File: app/views/hr_view.py (hunk 10 above)
Comment (Felixim0) [reply app/views/hr_view.py]: Removed

File: app/views/hr_view.py (hunk 4 above)
Comment (Felixim0) [reply app/views/hr_view.py]: removed

**Review by SammyPoot — APPROVED** [review]

**Pull request merged**